src/pipeline.ts

```diff
@@ -207,6 +211,26 @@ export class GitHubWorkflow extends PipelineBase {
     this.publishAssetsAuthRegion = props.publishAssetsAuthRegion ?? 'us-west-2';
   }
 
+  /**
+   * Parse AWS credential configuration from deprecated properties For backwards compatibility.
+   */
+  private getAwsCredentials(props: GitHubWorkflowProps) {
```

Comment:
Now that providing aws creds is convoluted with deprecated apis and such, I think it's necessary to have some checks here to make sure that the user is not supplying more than one of `githubActionRoleArn`, `awsCredentials`, or `awsCreds`.